pyinterprod/pronto/goa.py

```diff
@@ -234,3 +234,82 @@ def _get_constraints(term_id: str, ancestors: dict, constraints: dict) -> set:
         result |= _get_constraints(parent_id, ancestors, constraints)
 
     return result
+
+
+def import_go_constraints(go_url: str, pg_url: str):
+    pg_con = psycopg.connect(**url2dict(pg_url))
+    ora_con = oracledb.connect(go_url)
+    ora_cur = ora_con.cursor()
+
+    ora_cur.execute(
+        """
+        SELECT UNION_ID, LISTAGG(TAXON_ID, ',')
+        FROM GO.NCBI_TAXON_UNIONS
+        GROUP BY UNION_ID
+        """
+    )
+    union2taxon = {}
+    for union_id, taxon_ids in ora_cur:
+        union2taxon[union_id] = taxon_ids.split(",")
```

Comment:
Cast directly to a `set`:

```suggestion
        union2taxon[union_id] = set(taxon_ids.split(","))
```